@here/harp-mapview/lib/Tile.ts

```diff
@@ -949,994 +829,17 @@ export class Tile implements CachedResource {
-                    }
-
-                    count += groups[groupIndex].count;
-                }
-
-                const ObjectCtor = getObjectConstructor(technique);
-
-                if (ObjectCtor === undefined) {
-                    continue;
-                }
-
-                let material: THREE.Material | undefined = materials[techniqueIndex];
-
-                if (material === undefined) {
-                    const onMaterialUpdated = (texture: THREE.Texture) => {
-                        this.dataSource.requestUpdate();
-                        if (texture !== undefined) {
-                            this.addOwnedTexture(texture);
-                        }
-                    };
-                    material = createMaterial(
-                        {
-                            technique,
-                            level: Math.floor(displayZoomLevel),
-                            fog: this.mapView.scene.fog !== null
-                        },
-                        onMaterialUpdated
-                    );
-                    if (material === undefined) {
-                        continue;
-                    }
-                    materials[techniqueIndex] = material;
-                }
-
-                // Modify the standard textured shader to support height-based coloring.
-                if (
-                    isTerrainTechnique(technique) &&
-                    technique.heightBasedColors !== undefined &&
-                    technique.displacementMap !== undefined
-                ) {
-                    (material as any).onBeforeCompile = (shader: THREE.Shader) => {
-                        shader.fragmentShader = shader.fragmentShader.replace(
-                            "#include <map_pars_fragment>",
-                            `#include <map_pars_fragment>
-    uniform sampler2D displacementMap;
-    uniform float displacementScale;
-    uniform float displacementBias;`
-                        );
-
-                        shader.fragmentShader = shader.fragmentShader.replace(
-                            "#include <map_fragment>",
-                            `#ifdef USE_MAP
-    float minElevation = ${EarthConstants.MIN_ELEVATION.toFixed(1)};
-    float maxElevation = ${EarthConstants.MAX_ELEVATION.toFixed(1)};
-    float elevationRange = maxElevation - minElevation;
-
-    float disp = texture2D( displacementMap, vUv ).x * displacementScale + displacementBias;
-    vec4 texelColor = texture2D( map, vec2((disp - minElevation) / elevationRange, 0.0) );
-    texelColor = mapTexelToLinear( texelColor );
-    diffuseColor *= texelColor;
-#endif`
-                        );
-
-                        // We remove the displacement map from manipulating the vertices, it is
-                        // however still required for the pixel shader, so it can't be directly
-                        // removed.
-                        shader.vertexShader = shader.vertexShader.replace(
```

Comment:
This is what you are missing, see my other comment.

Reply:
That was helpful, I made sure to catch all changes in Tile.ts and moved them to TileGeometryCreator.ts.